File: frontend/src/types/api/financialAnalysisResponse.ts
import type { FinancialProfile } from '../enums/financialProfile'
import type {
  CategorySummary,
  CategoryPercentages,
  ClassifiedTransaction,
  FinancialMetrics,
  Recommendation,
} from './common'

export type FinancialAnalysisResponse = {
  financialProfile: FinancialProfile
  probability: number
  metrics: FinancialMetrics
  classifiedTransactions: ClassifiedTransaction[]
  categorySummary: CategorySummary
  categoryPercentages: CategoryPercentages
  recommendations: Recommendation[]
}

File: src/main/java/com/rendaflex/demo/dto/response/FinancialAnalysisResponse.java
package com.rendaflex.demo.dto.response;

import com.rendaflex.demo.enums.FinancialProfile;
import com.rendaflex.demo.enums.TransactionCategory;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

public record FinancialAnalysisResponse(
        FinancialProfile financialProfile,
        BigDecimal probability,
        FinancialMetrics metrics,
        List<ClassifiedTransaction> classifiedTransactions,
        Map<TransactionCategory, BigDecimal> categorySummary,
        Map<TransactionCategory, BigDecimal> categoryPercentages,
        List<Recommendation> recommendations
) {
}
